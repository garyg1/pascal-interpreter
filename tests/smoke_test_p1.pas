program smoke_test;

Var
    _yacc123_ : double;
    woke : double = 5.0;
    i : double = 0.123412341234;
    b : boolean;
    e : integer;

Const
    a : double = 1.0;
    c = false;

begin
    _yacc123_ := 11.1 + -2 * 5;
    _yacc123_ := _yacc123_ * woke;
    
    (* _yacc123_ = 1337; *)
    Writeln(_yacc123_);
    Writeln(woke);

    Readln(i);

    Writeln('math functions');
    Writeln(sqrt(i (* i am a comment! *)));
    Writeln(sin(i));
    Writeln(cos(i));
    Writeln(exp(i));
    Writeln(ln(i));

    Readln(i);

    writeln('logical statements');
    if (i > 10.0)
    then Writeln(i)
    else Writeln(a);

    e := trunc(i);

    if (a = 1) or not not (a = 1)
    then case e of
        1: Writeln(i * 100);
        2: WrItElN(i * 1000);
        else Writeln(a);
    end
    else Writeln(-1);
end.