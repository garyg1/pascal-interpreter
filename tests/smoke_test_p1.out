5.499999999999998
5.0
math functions
1.4142135623730951
0.9092974268256817
-0.4161468365471424
7.38905609893065
0.6931471805599453
logical statements
1.0
2000.0
